src/ai/backend/manager/data/agent/types.py

```diff
@@ -0,0 +1,138 @@
+from __future__ import annotations
+
+from collections.abc import Mapping
+from dataclasses import dataclass
+from datetime import datetime
+from decimal import Decimal
+from typing import TYPE_CHECKING, Any, Optional, Self
+
+from ai.backend.common.auth import PublicKey
+from ai.backend.common.types import AgentId, DeviceName, ResourceSlot, SlotName, SlotTypes
+from ai.backend.manager.models.agent import AgentStatus
+
+if TYPE_CHECKING:
+    from sqlalchemy.engine import Row
+
+
+@dataclass
+class AgentStateSyncData:
+    now: datetime
+    slot_key_and_units: dict[SlotName, SlotTypes]
+    current_addr: str
+    public_key: PublicKey
+
+
+@dataclass
+class AgentHeartbeatUpsert:
+    id: AgentId
+    status: AgentStatus
+    region: str
+    scaling_group: str
+    available_slots: ResourceSlot
+    addr: str
+    public_host: str
+    public_key: str
+    version: str
+    compute_plugins: Mapping[DeviceName, Mapping[str, str]]
+    architecture: str
+    auto_terminate_abusing_kernel: bool
+    lost_at: Optional[datetime]
+    first_contact: datetime
```

Comment:
There are too many fields in dataclass, could you please group the relevant data together?